### PR TITLE
adding package to install squashfs for singularity

### PR DESCRIPTION
**Pull request by vsoch**
This is my first contribution (thanks in advance for helping me learn the codebase so I can contribute more!) to add installation of the squashfs tools package in order to prepare for the Singularity upgrade.

 - I chose 4.3 as the default, the latest version
 - I didn't go earlier than 4.0 because it was like, back in 2009! :O 
 - the library only requires a make and make install, so I chose `MakefilePackage` as the base class (I found the package.py file in the main source folder) and then set the prefix to the prefix variable, which I figured out was provided in the function, and the makefile expected INSTALL_DIR to be set to the prefix.
 - to change directory, I found other examples `with working_dir` and that seemed to do the trick!

Let me know what needs to be changed / improved, etc., and glad to do it! I'll have a few more of these after this one, and hopefully learn a little more each time.

### CODE REVIEW

File: var/spack/repos/builtin/packages/squashfs/package.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+from spack import *
+
+class Squashfs(MakefilePackage):
+    """Squashfs - read only compressed filesystem"""
+
+    homepage = 'http://squashfs.sourceforge.net'
+    url      = 'https://downloads.sourceforge.net/project/squashfs/squashfs/squashfs4.3/squashfs4.3.tar.gz'
+
+    # version      md5sum
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/squashfs/package.py]: Nowadays we prefer sha256 over md5. At some point we plan on updating all the older packages, but for new packages we prefer to use sha256.

Comment (vsoch) [reply var/spack/repos/builtin/packages/squashfs/package.py]: Is sha1 okay? Looks like squashfs provides that along with md5:

image

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/squashfs/package.py]: sha1 is certainly better than md5. You can also download, verify the sha1/md5, then compute the sha256.

On macOS:
```console
$ shasum -a 256 squashfs4.3.tar.gz
```
On Linux:
```console
$ sha256sum squash4.3.tar.gz
```
If it was any other website than SourceForge, you could run `spack checksum` to do all of this automatically, but in my experience SourceForge prevents web crawlers.

Comment (vsoch) [reply var/spack/repos/builtin/packages/squashfs/package.py]: Looks like it's ok?

```python
#: Set of hash algorithms that Spack can use, mapped to digest size in bytes
hashes = {
    'md5': 16,
    'sha1': 20,
    'sha224': 28,
    'sha256': 32,
    'sha384': 48,
    'sha512': 64
}
```

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/squashfs/package.py]: Yep, Spack can handle any hashing algorithm, although sha256 is the current default when you create a new package/add a new version.

Comment (vsoch) [reply var/spack/repos/builtin/packages/squashfs/package.py]: okay all set! I just pushed sha1, and I'm running locally to test again.

**Comment by vsoch** [comment]
:green_apple: :green_book: :green_heart: :green_salad: 

Quick unrelated question - if a library is already added (libuuid for example) does that also cover libuuid-dev since the source code is included (and the libraries are added to some path?)

**Comment by adamjstewart** [comment]
> if a library is already added (libuuid for example) does that also cover libuuid-dev since the source code is included (and the libraries are added to some path?)

Yes, unix package managers often split the libraries/binaries and headers into two separate packages (libuuid and libuuid-dev) but Spack installs everything together.

**Comment by vsoch** [comment]
cool, thanks.

**Pull request merged**